extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergManualRefreshTableLocationProvider.java

```diff
@@ -58,62 +52,29 @@ public String getImplementationName() {
 
     @Override
     public synchronized void refresh() {
-        // There should be no refresh service for this provider.
-        throw new UnsupportedOperationException();
+        // There is no refresh service for this provider, but this is called as part of the initialization process.
+        refreshLocations();
```

Comment:
I think the better pattern might be to override `doInitialize()` -> `refreshLocations()`, and have this be a call to `ensureInitialized()`. This is just to make sure we properly pick up the first snapshot, since subsequent changes are delivered when the user asks for them.

It might be more appropriate to make `doInitialize()` `synchronized`, instead of `refresh()`, but we seem to be allowing `ensureInitialized()` to race internally.